server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/DoubleValuesComparatorSource.java

```diff
@@ -98,11 +94,11 @@ private NumericDoubleValues getNumericDocValues(LeafReaderContext context, doubl
     protected void setScorer(Scorable scorer, LeafReaderContext context) {}
 
     @Override
-    public FieldComparator<?> newComparator(String fieldname, int numHits, boolean enableSkipping, boolean reversed) {
+    public FieldComparator<?> newComparator(String fieldname, int numHits, Pruning pruning, boolean reversed) {
         assert indexFieldData == null || fieldname.equals(indexFieldData.getFieldName());
 
         final double dMissingValue = (Double) missingObject(missingValue, reversed);
-        return new DoubleComparator(numHits, fieldname, dMissingValue, reversed, enableSkipping && this.enableSkipping) {
+        return new DoubleComparator(numHits, fieldname, dMissingValue, reversed, pruning) {
```

Comment:
I think if skipping has been disabled a the `ComparatorSource` level (via `disableSkipping`), we need to ignore the passed value.

Maybe in `XFieldComparatorSource` it makes sense to add a little helper method like:

```
protected Pruning filterPruning(Pruning proposedPruning) {
  if (this.pruning == Pruning.NONE) {
    return Pruning.NONE;
  }
  return proposedPruning.
}
```

Then each of these calls can "filter" the proposed pruning values. 

Actually, I would consider leaving the `XFieldComparatorSource`'s `enableSkipping` field there as a `boolean`, and the `filterPruning` method would become:

```
protected Pruning filterPruning(Pruning proposedPruning) {
  if (this.enableSkipping) {
    return proposedPruning;
  }
  return Pruning.NONE.
}
```

Note that the logic around disabling skipping at the `ComparatorSource` level was added in https://github.com/opensearch-project/OpenSearch/pull/8168.

Reply:
Makes sense @msfroh , keeping enableSkipping as it is and passing proposedPruning in helper method `filterPruning` is good idea.

Reply:
Updated this in latest revision